operator/api/redpanda/v1alpha3/redpanda_conversion.go

```diff
@@ -0,0 +1,954 @@
+				panic(err)
+			}
+			return value, true
+		}))
+
+		var image *redpandav1alpha2.RedpandaImage
+		if idx := strings.LastIndex(src.Spec.NodePoolSpec.BrokerTemplate.Image, ""); idx != -1 {
+			image = &redpandav1alpha2.RedpandaImage{
+				Repository: ptr.To(src.Spec.NodePoolSpec.BrokerTemplate.Image[:idx]),
+				Tag:        ptr.To(src.Spec.NodePoolSpec.BrokerTemplate.Image[idx:]),
+				PullPolicy: nil,
+			}
+		} else if src.Spec.NodePoolSpec.BrokerTemplate.Image != "" {
+			image = &redpandav1alpha2.RedpandaImage{
+				Repository: ptr.To(src.Spec.NodePoolSpec.BrokerTemplate.Image),
+				Tag:        nil,
+				PullPolicy: nil,
+			}
+		}
+
+		return &redpandav1alpha2.RedpandaClusterSpec{
+			Image:         image,
+			Enterprise:    convertTo[*redpandav1alpha2.Enterprise](src).(*redpandav1alpha2.Enterprise),
+			ClusterDomain: nilIfZero(src.Spec.ClusterDomain),
+			Config: &redpandav1alpha2.Config{
+				RPK:  rpkConfig,
+				Node: nodeConfig,
+
+				// All ClusterConfig gets pushed into ExtraClusterConfiguration
+				Tunable:                   nil,
+				Cluster:                   nil,
+				ExtraClusterConfiguration: convertTo[redpandav1alpha2.ClusterConfiguration](src).(redpandav1alpha2.ClusterConfiguration),
+
+				// Unused fields
+				SchemaRegistryClient: nil,
+				PandaProxyClient:     nil,
+			},
+			TLS:         convertTo[*redpandav1alpha2.TLS](src).(*redpandav1alpha2.TLS),
+			Listeners:   convertTo[*redpandav1alpha2.Listeners](src).(*redpandav1alpha2.Listeners),
+			Statefulset: convertTo[*redpandav1alpha2.Statefulset](src).(*redpandav1alpha2.Statefulset),
+			Resources: &redpandav1alpha2.Resources{
+				Limits:   (*map[corev1.ResourceName]resource.Quantity)(&src.Spec.NodePoolSpec.BrokerTemplate.Resources.Limits),
+				Requests: (*map[corev1.ResourceName]resource.Quantity)(&src.Spec.NodePoolSpec.BrokerTemplate.Resources.Requests),
+				CPU:      nil,
+				Memory:   nil,
+			},
+
+			// TODO Audit this list.
+			Affinity:                   nil,
+			AuditLogging:               nil,
+			Auth:                       nil,
+			CommonLabels:               nil,
+			Connectors:                 nil,
+			Console:                    nil,
+			DeprecatedFullNameOverride: "",
+			External:                   nil,
+			Force:                      nil,
+			FullnameOverride:           nil,
+			ImagePullSecrets:           nil,
+			LicenseKey:                 nil,
+			LicenseSecretRef:           nil,
+			Logging:                    nil,
+			Monitoring:                 nil,
+			NameOverride:               nil,
+			NodeSelector:               nil,
+			PostInstallJob:             nil,
+			PostUpgradeJob:             nil,
+			RBAC:                       nil,
+			RackAwareness:              nil,
+			Service:                    nil,
+			ServiceAccount:             nil,
+			Storage:                    nil,
+			Tests:                      nil,
+			Tolerations:                nil,
+			Tuning:                     nil,
```

Comment:
So, if we want to keep these convertible without losing info, but also dropping support (though pretty sure some will be in the pod template), could also do the whole marshal onto an annotation thing.